### PR TITLE
Add delete user endpoint

### PR DESCRIPTION
**Pull request by tfnribeiro**
+ Expose a way to delete users to the API
+ Uses the user current session id to start the deletion process. 

Used by: https://github.com/zeeguu/web/pull/373

### CODE REVIEW

File: zeeguu/api/endpoints/accounts.py [hunk 1]
```diff
@@ -13,6 +13,25 @@
 from zeeguu.logging import log
 
 
+@api.route("/delete_user/<session_id>", methods=["POST"])
+@cross_domain
+def remove_user(session_id):
+    from zeeguu.core.account_management.user_account_deletion import delete_user_account
+
+    try:
+        session_id = int(session_id)
```
Comment (mircealungu) [inline zeeguu/api/endpoints/accounts.py:22]: why do we pass the session_id in the URL? Can't we use the @with_session decorator?

**Review by mircealungu — CHANGES_REQUESTED** [review]
Tiago: one more observation on this PR

File: zeeguu/api/endpoints/accounts.py (hunk 1 above)
Comment (tfnribeiro) [reply zeeguu/api/endpoints/accounts.py:22]: That's a good point -  I will check if it works using the decorator!

**Comment by tfnribeiro** [comment]
I have changed to use the session. I also added a test case for the user deletion. I remember we talked about a possible issue with ordering, but I don't recall the solution? Right now, it is at the end of all the `user_tests`, where it deletes the `annonymous` user.

File: zeeguu/core/account_management/user_account_deletion.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    User,
+    UserActivityData,
+    Bookmark,
+    UserArticle,
+    UserReadingSession,
+    UserExerciseSession,
+)
+from zeeguu.core.model import Article
+
+tables_to_modify = [
```
Comment (mircealungu) [inline zeeguu/core/account_management/user_account_deletion.py:33]: I remember seeing this elsewhere. Ah, yes, in `delete_dev_users`. Would it be possible to remove it from there, and thus, remove the duplication between the two files. I think also the `delete_user_account` method is duplicated with `delete_user`?

**Review by mircealungu — CHANGES_REQUESTED** [review]
small favor to ask :)

**Pull request merged**